tests/Unit.hs

```diff
@@ -35,14 +35,21 @@ legacyEncoding = False
 --   whereas legacy will immediately stop.
 whenLegacy s = if legacyEncoding then s else B.empty
 
+toTest = do
+    args <- getArgs
+    return $ case args of [] -> ["cabal", "run", "-v0", "--ghc-options=-w", "--"]
```

Comment:
As you mentioned, this is pretty hacky.  Can we replace this by using the `Paths_haskeline` module that Cabal auto-generates (or something similar), so that we can run the binary directly?